tests/observability/runbook_url/test_runbook_url.py

```diff
@@ -1,13 +1,37 @@
+import http
 import logging
 
 import pytest
+import requests
 
-from tests.utils import validate_runbook_url_exists
-from utilities.constants import CNV_PROMETHEUS_RULES, QUARANTINED
+from utilities.constants import CNV_PROMETHEUS_RULES
 
 LOGGER = logging.getLogger(__name__)
 
 
+def validate_downstream_runbook_url(runbook_urls_from_prometheus_rule: dict[str, str]) -> None:
+    error_messages = {}
+    alerts_without_runbook = []
+
+    for alert_name, runbook_url in runbook_urls_from_prometheus_rule:
+        if not runbook_url:
+            LOGGER.error(f"For alert: {alert_name} Url not found")
+            alerts_without_runbook.append(alert_name)
+            continue
+        try:
+            response = requests.get(runbook_url, allow_redirects=False, timeout=10)
+            if response.status_code != http.HTTPStatus.OK:
+                LOGGER.error(f"Alert {alert_name} url {runbook_url} returned status {response.status_code}")
+                error_messages[alert_name] = runbook_url
+        except requests.RequestException as e:
+            LOGGER.error(f"Alert {alert_name} url {runbook_url} failed: {e}")
+            error_messages[alert_name] = runbook_url
+    assert not (alerts_without_runbook or error_messages), (
+        f"CNV alerts with missing runbook url: {alerts_without_runbook}, "
+        f"D/S runbook url validation failed for the followings alerts: {error_messages}"
+    )
```

Comment:
What do you think about trying to use `subtests`? it will make the code shorter and more simple to read
```suggestion
    for alert_name, runbook_url in runbook_urls_from_prometheus_rule:
        with request.subtests.test(alert_name=alert_name):
            assert runbook_url, f"For alert: {alert_name} Url not found"
            try:
                response = requests.get(runbook_url, allow_redirects=False, timeout=10)
                assert response.status_code == http.HTTPStatus.OK, (
                    f"Alert {alert_name} url {runbook_url} returned status {response.status_code}"
                )
            except requests.RequestException as e:
                pytest.fail(f"Alert {alert_name} url {runbook_url} failed: {e}")
```

Reply:
Done